src/wallet/wallet.cpp

```diff
@@ -1359,6 +1359,308 @@ void CWallet::SyncMetaData(pair<typename TxSpendMap<T>::iterator, typename TxSpe
+                if (accountId.has_value()) {
+                    pattern = AccountZTXOPattern(accountId.value(), ua.GetKnownReceiverTypes());
+                }
+            }
+        }
+    }, addr);
+
+    if (pattern.has_value()) {
+        return ZTXOSelector(pattern.value(), requireSpendingKey);
+    } else {
+        return std::nullopt;
+    }
+}
+
+ZTXOSelector CWallet::LegacyTransparentZTXOSelector() {
+    return ZTXOSelector(
+            AccountZTXOPattern(ZCASH_LEGACY_ACCOUNT, {ReceiverType::P2PKH, ReceiverType::P2SH}),
+            true);
+}
+
+std::optional<libzcash::AccountId> CWallet::FindAccountForSelector(const ZTXOSelector& selector) const {
+    auto self = this;
+    std::optional<libzcash::AccountId> result{};
+    std::visit(match {
+        [&](const CKeyID& addr) {
+            auto meta = self->GetUFVKMetadataForReceiver(addr);
+            if (meta.has_value()) {
+                result = self->GetUnifiedAccountId(meta.value().first);
+            }
+        },
+        [&](const CScriptID& addr) {
+            auto meta = self->GetUFVKMetadataForReceiver(addr);
+            if (meta.has_value()) {
+                result = self->GetUnifiedAccountId(meta.value().first);
+            }
+        },
+        [&](const libzcash::SproutPaymentAddress& addr) { },
+        [&](const libzcash::SaplingPaymentAddress& addr) {
+            auto meta = GetUFVKMetadataForReceiver(addr);
+            if (meta.has_value()) {
+                result = self->GetUnifiedAccountId(meta.value().first);
+            }
+        },
+        [&](const AccountZTXOPattern& acct) {
+            if (self->mnemonicHDChain.has_value() &&
+                self->mapUnifiedAccountKeys.count(std::make_pair(self->mnemonicHDChain.value().GetSeedFingerprint(), acct.GetAccountId())) > 0) {
+                result = acct.GetAccountId();
+            }
+        }
+    }, selector.GetPattern());
+    return result;
+}
+
+bool CWallet::SelectorMatchesAddress(
+        const ZTXOSelector& selector,
+        const CTxDestination& address) const {
+    auto self = this;
+    return std::visit(match {
+        [&](const CKeyID& keyId) {
+            CTxDestination keyIdDest = keyId;
+            return address == keyIdDest;
+        },
+        [&](const CScriptID& scriptId) {
+            CTxDestination scriptIdDest = scriptId;
+            return address == scriptIdDest;
+        },
+        [&](const libzcash::SproutPaymentAddress& addr) { return false; },
+        [&](const libzcash::SaplingPaymentAddress& addr) { return false; },
+        [&](const AccountZTXOPattern& acct) {
+            if (acct.IncludesP2PKH() || acct.IncludesP2SH()) {
+                std::optional<std::pair<libzcash::UFVKId, std::optional<libzcash::diversifier_index_t>>> meta;
+                std::visit(match {
+                    [&](const CNoDestination& none) { meta = std::nullopt; },
+                    [&](const auto& addr) { meta = self->GetUFVKMetadataForReceiver(addr); }
```

Comment:
Spell out this type; I think it is `const libzcash::UFVKId&` but I had to check the header.

Reply:
What are you referring to? The type is explicitly written out (laboriously) on line 1469.